code/espurna/config/hardware.h

```diff
@@ -3103,6 +3103,51 @@
     #define RELAY2_PIN          5
     #define RELAY2_TYPE         RELAY_TYPE_NORMAL
 
+#elif defined(ALLTERCO_SHELLY1PM)
+    // Info
+    #define MANUFACTURER        "ALLTERCO"
+    #define DEVICE              "SHELLY1PM"
+
+    // Buttons
+    #define BUTTON1_PIN         4
+    #define BUTTON1_MODE        BUTTON_SWITCH
+    #define BUTTON1_RELAY       1
+
+    #define BUTTON2_PIN         2
+    #define BUTTON2_MODE        BUTTON_PUSHBUTTON | BUTTON_DEFAULT_HIGH
+    #define BUTTON2_LNGCLICK    BUTTON_MODE_RESET
+    #define BUTTON2_LNGLNGCLICK BUTTON_MODE_FACTORY
+
+    // Relays
+    #define RELAY1_PIN          15
+    #define RELAY1_TYPE         RELAY_TYPE_NORMAL
+
+    // Light
+    #define LED1_PIN            0
+    #define LED1_PIN_INVERSE    1
+
+    // HJL01 / BL0937
+    #ifndef HLW8012_SUPPORT
+    #define HLW8012_SUPPORT             1
+    #endif    
+    #define HLW8012_SEL_PIN             12
+    #define HLW8012_CF1_PIN             13
```

Comment:
Comment here too that they are unused?

Reply:
yes, they are unused but I didn't know if I comment that then It will take another default pin that we don't want. If espurna will take default values for those SEL and CF1, we need to change it to the default unused pins we want, am I correct?

Here is an example: [SEL](https://github.com/xoseperez/espurna/blob/868d4d5573cabb33b3d50f88f7ff4ac2dd5b56db/code/espurna/config/sensors.h#L440) is by default one of the currently used pins

Reply:
Yeah, defaults are copying the first Sonoff POW, so something must be there. I was thinking of adding a comment, not removing the pin definitions. But, I think you can use `GPIO_NONE` (0x99) instead of the real ones. Arduino layer for digitalWrite, pinMode, attachInterrupt etc. has these two kinds of checks:
```
func(pin) {
  if(pin < 16){
  ...
  } else if(pin == 16){
  ...
  }
}
```
Or just `pin < 16`, so they will do nothing

ref:
https://github.com/esp8266/Arduino/blob/2.3.0/cores/esp8266/core_esp8266_wiring_digital.c
https://github.com/esp8266/Arduino/blob/master/cores/esp8266/core_esp8266_wiring_digital.cpp

Reply:
Wow I didn't know that, I will change it for `GPIO_NONE`